librepo/handle.c

```diff
@@ -264,6 +266,33 @@ lr_handle_remote_sources_changed(LrHandle *handle, LrChangedRemoteSource type)
     }
 }
 
+struct callback_data {
+   GMainLoop *loop; 
+   long input_seconds;
+   time_t begin;
+   GNetworkMonitor *monitor;
+   GSocketConnectable *connectable; 
+   GCancellable *cancellable;
+};
+
+gboolean timeout_callback(gpointer data)
+{ 
+  struct callback_data *dt = (struct callback_data*)data;
+   if(!g_network_monitor_can_reach(dt->monitor, dt->connectable, dt->cancellable, NULL) &&
+      time(NULL) - dt->begin <= dt->input_seconds){
+      return TRUE;
+   } 
+   g_main_loop_quit(dt->loop);
+   return FALSE;
+}
+
+void on_network_available(GObject *object, GParamSpec *pspec, gpointer data)
```

Comment:
When we get notified about this, it may actually be because the property transitioned from `true` :arrow_right: `false`!  So...related to my comment below, I think to be a bit more precise here we should call `g_network_monitor_get_network_available ()` - and start/stop polling the URI based on that.